WalletWasabi.Daemon/Rpc/WasabiJsonRpcService.cs

```diff
@@ -169,6 +145,51 @@ public JsonRpcResult WalletInfo()
 		return info;
 	}
 
+	[JsonRpcMethod("getcoinjoininfo")]
+	public JsonRpcResult GetCoinjoinInfo(string? password = null)
+	{
+		var coinJoinManager = Global.HostedServices.Get<CoinJoinManager>();
+		var activeWallet = Guard.NotNull(nameof(ActiveWallet), ActiveWallet);
+		var serverTipHeight = activeWallet.BitcoinStore.SmartHeaderChain.ServerTipHeight;
+
+		AssertWalletIsLoaded();
+		AssertWalletIsLoggedIn(activeWallet, password ?? "");
+
+		var coinsInCoinjoin = activeWallet.Coins.Where(coin => coin.CoinJoinInProgress);
+
+		var currentRounds = coinJoinManager.RoundStatusUpdater.RoundStates;
+		var roundsIdsWithMyInputs = currentRounds.Values.ToDictionary(round => round.Id, _ => new List<JsonRpcResult>());
+
+		var coinsInInputRegistration = new List<JsonRpcResult>();
+		foreach (var coinInCoinjoin in coinsInCoinjoin)
+		{
+			var roundRegisteredTo =
+				currentRounds.Values.FirstOrDefault(x =>
+					x.CoinjoinState.Inputs.Any(y => y.Outpoint == coinInCoinjoin.Outpoint));
+
+			// Coin in coinjoin but not registered to any round, must be in InputRegistration
+			// We can't know in which round it will be registered
+			if (roundRegisteredTo is null)
+			{
+				coinsInInputRegistration.Add(GetRpcResultFromSmartCoin(coinInCoinjoin, serverTipHeight));
+				continue;
+			}
+
+			roundsIdsWithMyInputs[roundRegisteredTo.Id].Add(GetRpcResultFromSmartCoin(coinInCoinjoin, serverTipHeight));
+		}
+
+		return new JsonRpcResult
+		{
+			["coinsRegistering"] = coinsInInputRegistration.ToImmutableArray(),
+			["roundsWithMyInputsRegistered"] = roundsIdsWithMyInputs.Select(
```

Comment:
Variable name is hard for understanding. Possible naming:
```suggestion
			["roundsWithMyInputsRegistered"] = idRoundsWithMyInput.Select(
```